frontend/src/views/components/Map/Map.component.tsx

```diff
@@ -63,10 +72,22 @@ const MapView = (props: MapViewProps) => {
   const { query, locale } = router;
   const { setParam } = useRouterParams();
   const layerManagerRef = useContext(LayerManagerContext);
-
+  const subdomainIsDifferentThanLoaded =
+    layersLoadedSubdomain !== subdomain && (layersLoadedSubdomain || subdomain);
+  const layerGroupsSubdomainIsDifferentThanLoaded =
+    layerGroupsLoadedSubdomain !== subdomain && (layerGroupsLoadedSubdomain || subdomain);
   useEffect(() => {
-    if (!layersLoaded || layersLoadedLocale !== locale) loadLayers(locale);
-    if (!layerGroupsLoaded || layerGroupsLoadedLocale !== locale) loadLayerGroups(locale);
+    if (!layersLoaded || layersLoadedLocale !== locale || subdomainIsDifferentThanLoaded) {
+      loadLayers(locale);
+    }
+
+    if (
+      !layerGroupsLoaded ||
+      layerGroupsLoadedLocale !== locale ||
+      layerGroupsSubdomainIsDifferentThanLoaded
+    ) {
+      loadLayerGroups(locale);
+    }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [locale]);
```

Comment:
You should always include all the dependencies, here the variables that you created.